msglibrary/src/test/java/ingsw/group1/msglibrary/SMSPeerTest.java

```diff
@@ -18,113 +19,74 @@
  * @author Giorgia Bortoletti
  * @author Riccardo De Zen
  */
-@Config(sdk = Build.VERSION_CODES.P)
-@RunWith(RobolectricTestRunner.class)
+@RunWith(Parameterized.class)
 public class SMSPeerTest {
 
-    private SMSPeer peer;
-    private static final String EX_VALID_ADDRESS = "+39892424";
-
-    @Test
-    public void isAddressValidNull(){
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid(null)
-        );
-    }
-
-    @Test
-    public void isAddressValidEmptyOrBlank(){
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid("")
-        );
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid("     ")
-        );
-    }
-
-    @Test
-    public void isAddressValidNaN(){
-        String notANumber = "Wassup";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid(notANumber)
-        );
-    }
-
-    @Test
-    public void isAddressValidNoCountryCode(){
-        String noCC = "911";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NO_COUNTRY_CODE,
-                SMSPeer.isAddressValid(noCC)
-        );
+    public static final int TEST_RUNS = 100;
+    private final int run;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'run' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-message-library/pullRequest?prid=4741620)